### ISSUE TITLE
Is it possible to specify different weights in the loss computation for training input in different folders?

### ISSUE THREAD

**Issue by sanjaykasturia** [facebookresearch/wav2letter#396]
Train allows you to list multiple folders for the input training data. In the loss computation, it would be useful to allow different weights for training inputs from different folders. 
Main purpose for this would be to allow a quick adjustment to a previously trained model to fix specific weakeness by providing additional training data targeted to fixing the problem. 
When adding additional targetted training data, it would be nice to have a way to attach a higher weight to it in the loss computation. This will compensate for the fact that the additional training data may have a lot fewer samples that the main training corpus. This can also be used to personalize a pretrained acoustic model to a specific user by adding a smaller amount of labeled speech from the specific user and reconverging the network with a higher weight on the user specific data.

**Comment by tlikhomanenko** [comment]
Hi @sanjaykasturia,
Will consider extending the functionality for this.

For now you could fix this in a quick way for your experiments by adding weight coefficient based on `sample[kSampleIdx]` here https://github.com/facebookresearch/wav2letter/blob/master/Train.cpp#L536. 

If you have specific weight for each audio sample then you could provide additional field here https://github.com/facebookresearch/wav2letter/blob/master/src/data/Utils.h#L21 and reading the weights from the file (fix here https://github.com/facebookresearch/wav2letter/blob/master/src/data/W2lListFilesDataset.cpp#L118). Then use this additionally introduced field while computing the loss https://github.com/facebookresearch/wav2letter/blob/master/Train.cpp#L536.

**Comment by tlikhomanenko** [comment]
No plans to have this, but it simply can be done with modifying train.cpp.
